cdap-ui/app/cdap/components/PipelineConfigurations/ConfigurationsContent/ComputeTabContent/index.js

```diff
@@ -14,15 +14,65 @@
  * the License.
 */
 
+import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-import ProfilesListViewInPipeline from 'components/PipelineConfigurations/ConfigurationsContent/ComputeTabContent/ProfilesListView';
+import ProfilesListViewInPipeline, {PROFILE_NAME_PREFERENCE_PROPERTY} from 'components/PipelineConfigurations/ConfigurationsContent/ComputeTabContent/ProfilesListView';
+import PipelineConfigurationsStore, {ACTIONS as PipelineConfigurationsActions} from 'components/PipelineConfigurations/Store';
+import findIndex from 'lodash/findIndex';
+import uuidV4 from 'uuid/v4';
+import {updatePipelineEditStatus} from 'components/PipelineConfigurations/Store/ActionCreator';
+import {connect} from 'react-redux';
+import {objectQuery} from 'services/helpers';
 
-export default class ComputeTabContent extends Component {
+class ComputeTabContent extends Component {
+
+  static propTypes = {
+    selectedProfile: PropTypes.object
+  };
+
+  onProfileSelect = (profileName) => {
+    let {runtimeArgs} = PipelineConfigurationsStore.getState();
+    let pairs = [...runtimeArgs.pairs];
+    pairs = pairs.filter(pair => pair.key.length);
```

Comment:
is there a case where the store will have some value that has empty key? I thought we won't add those to the store